presidio-anonymizer/presidio_anonymizer/entities/engine/dict_recognizer_result.py

```diff
@@ -0,0 +1,29 @@
+from dataclasses import dataclass
+from typing import List, Union, Iterator
+
+from presidio_anonymizer.entities import RecognizerResult
+
+
+@dataclass
+class DictRecognizerResult:
+    """
+    Data class for holding the output of the Presidio Analyzer on dictionaries.
+
+    :param key: key in dictionary
+    :param value: value to run analysis on (either string or list of strings)
+    :param recognizer_results: Analyzer output for one value.
+    Could be either:

```

Comment:
as a client/user it is a bit confusing to be able to provide different types to the same data structure. maybe its not the same type or some sort of inheritance?

Reply:
to keep the implemetation simple, I did not make any changes to the object created in the analyzer package DictAnalyzerResult. I'll consider changig it

Reply:
Normally the user wouldn't interact with this object, as it is meant to pass analyzer results to the anonymizer. The reason it accepts multiple types is because of the json nature which can either be a string or a list.